File: cakk-api/src/main/java/com/cakk/api/dto/request/shop/CreateShopRequest.java
package com.cakk.api.dto.request.shop;

import java.util.List;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import com.cakk.api.annotation.OperationDay;
import com.cakk.core.dto.param.shop.ShopLinkParam;
import com.cakk.core.dto.param.shop.ShopOperationParam;


public record CreateShopRequest(
	@Size(max = 20)
	String businessNumber,
	@OperationDay
	List<ShopOperationParam> operationDays,
	@NotBlank @Size(max = 30)
	String shopName,
	@Size(max = 40)
	String shopBio,
	@Size(max = 500)
	String shopDescription,
	@NotBlank @Size(max = 50)
	String shopAddress,
	@NotNull @Min(-90) @Max(90)
	Double latitude,
	@NotNull @Min(-180) @Max(180)
	Double longitude,
	@NotNull
	List<ShopLinkParam> links
) {
}

File: cakk-api/src/main/kotlin/com/cakk/api/dto/request/shop/CreateShopRequest.kt
package com.cakk.api.dto.request.shop

import jakarta.validation.constraints.*

import com.cakk.api.annotation.OperationDay
import com.cakk.core.dto.param.shop.ShopLinkParam
import com.cakk.core.dto.param.shop.ShopOperationParam

data class CreateShopRequest(
	@field:Size(max = 20)
    val businessNumber: String?,
    @field:OperationDay
	val operationDays: List<ShopOperationParam>?,
	@field:NotBlank @field:Size(max = 30)
    val shopName: String?,
	@field:Size(max = 40)
    val shopBio:  String?,
	@field:Size(max = 500)
    val shopDescription:  String?,
	@field:NotBlank @field:Size(max = 50)
    val shopAddress:  String?,
	@field:NotNull @field:Min(-90) @field:Max(90)
    val latitude:  Double?,
	@field:NotNull @field:Min(-180) @field:Max(180)
    val longitude:  Double?,
	@field:NotNull
	val links: List<ShopLinkParam>?
)
